src/test-run/index.js

```diff
@@ -155,10 +157,44 @@ export default class TestRun extends Session {
     }
 
     _resolvePendingRequest (command) {
+        this.lastDriverStatusResponse = command;
         this.pendingRequest.resolve(command);
         this.pendingRequest = null;
     }
 
+    _handleDriverRequest (driverStatus) {
+        if (!this.running)
+            this._start();
+
+        var shouldRejectPendingDriverTask = driverStatus.pageError &&
+                                            this.pendingDriverTask &&
+                                            isCommandRejectableByPageError(this.pendingDriverTask.command);
+
+        if (shouldRejectPendingDriverTask)
+            this._rejectPendingDriverTask(driverStatus.pageError);
+        else {
+            this.pendingJsError = this.pendingJsError || driverStatus.pageError;
+
+            if (this.pendingDriverTask) {
+                if (!driverStatus.isCommandResult)
+                    return this.pendingDriverTask.command;
+
+                if (isTestDoneCommand(this.pendingDriverTask.command)) {
+                    this.pendingDriverTask.resolve();
+
+                    return {};
```

Comment:
Can we move it to const, smth like `TestDoneConfirmationCommand` or smthg. Because it's a little bit confusing, because function has too much return types: empty object, command, null.